lms/djangoapps/course_home_api/outline/v1/views.py

```diff
@@ -21,13 +24,24 @@
 from lms.djangoapps.courseware.date_summary import TodaysDate
 from lms.djangoapps.courseware.masquerade import setup_masquerade
 from openedx.core.djangoapps.content.block_structure.transformers import BlockStructureTransformers
+from openedx.core.djangoapps.user_api.course_tag.api import get_course_tag
+from openedx.features.course_experience import COURSE_ENABLE_UNENROLLED_ACCESS_FLAG, LATEST_UPDATE_FLAG
 from openedx.features.course_experience.course_tools import CourseToolsPluginManager
-from openedx.features.course_experience import COURSE_ENABLE_UNENROLLED_ACCESS_FLAG
+from openedx.features.course_experience.views.latest_update import LatestUpdateFragmentView
+from openedx.features.course_experience.views.welcome_message import (
+    dismiss_welcome_message as disable_welcome_message, PREFERENCE_KEY, WelcomeMessageFragmentView
+)
 from student.models import CourseEnrollment
 from xmodule.course_module import COURSE_VISIBILITY_PUBLIC
 from xmodule.modulestore.django import modulestore
 
 
+class UnableToDismissWelcomeMessage(APIException):
+    status_code = 400
+    default_detail = 'Unable to dismiss welcome message.'
+    default_code = 'unable_to_dismiss_welcome_message'
```

Comment:
Nit: I feel like you could just return a Response with this information rather than making a new Exception class that is only used in one spot. Up to you.